### PR TITLE
PID Control Without UI

### PR DESCRIPTION
**Pull request by IrvinRamiA**
#26

### CODE REVIEW

File: Code/Integrator_Project/src/InputCapture.c [hunk 1]
```diff
@@ -1,45 +1,49 @@
...
 
 void InputCapture_Callback(input_capture_callback_args_t *p_args)
 {
-    uint32_t pclkdFreq = 0;
-    uint64_t timeCaptured = 0;
+    uint32_t pclkdFreq = ZeroInputCapture;
+    uint64_t timeCaptured = ZeroInputCapture;
 
-    if(CHANNEL_ZERO == p_args->channel)
+    if(CHANNEL_TWO == p_args->channel)
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/InputCapture.c:31]: Why have you changed channel zero to two?

File: Code/Integrator_Project/src/PidControl.c [hunk 2]
```diff
@@ -7,28 +7,65 @@
...
     currentTimeInCycles = DWT->CYCCNT;
     currentTimeInMs = currentTimeInCycles / CyclesPerMs;
     timeChangeInMs = currentTimeInMs - lastTimeInMs;
+
     if(timeChangeInMs >= SampleTimeInMs)
     {
-        error = (uint8_t) (*mySetPointInRpm - *myMeasuredSpeedInRpm);
-        derivativeError = (uint8_t) (error - lastError);
-        integralError = (uint8_t) (integralError + error);
+        if(*mySetPointInRpm >= *myMeasuredSpeedInRpm){
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/PidControl.c:49]: The braces is in same line of "if"

File: Code/Integrator_Project/src/PidControl.h [hunk 3]
```diff
@@ -10,30 +10,44 @@
 
 #include "hal_data.h"
 
+uint32_t error;
+uint32_t derivativeError;
+uint32_t integralError;
+uint32_t lastError;
+uint32_t outputDebug;
+
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/PidControl.h:18]: EWS?

File: Code/Integrator_Project/src/PidControl.h [hunk 4]
```diff
@@ -10,30 +10,44 @@
...
+uint32_t currentTimeInMs;
+uint32_t currentTimeInCycles;
+uint32_t lastTimeInMs;
+uint32_t timeChangeInMs;
+double sampleTimeInSec;
+
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/PidControl.h:24]: Same here.

File: Code/Integrator_Project/src/PidControl.h [hunk 5]
```diff
@@ -10,30 +10,44 @@
...
+double kP;
+double kI;
+double kD;
+
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/PidControl.h:28]: Same here.

File: Code/Integrator_Project/src/hal_entry.c [hunk 6]
```diff
@@ -10,29 +10,30 @@
...
-    OpenAdc();
-    ScanCfgAdc();
-    ScanStartAdc();
-
-    OpenTimer();
-    StartTimer();
-
+    InitializeAdc();
+    InitializePwm();
     InitializeInputCapture();
```
Comment (JuanLNH) [inline Code/Integrator_Project/src/hal_entry.c:26]: EWS

**Review by JuanLNH — APPROVED** [review]
LGTM!

File: Code/Integrator_Project/src/InputCapture.c (hunk 1 above)
Comment (IrvinRamiA) [reply Code/Integrator_Project/src/InputCapture.c:31]: Because the channel 0 pin is now used by the **Display**

**Pull request merged**